### PR TITLE
Making the cross plat code release ready.

### PR DESCRIPTION
**Pull request by ankitgo**
1) Making sure that timeline apis hit the latest controllers of DT and not the compatible controllers.
2) Adding isolation logic on the basic of 'system' variable.
3) Adding a download artifact before job plugin for release.

### CODE REVIEW

**Comment by bryanmacfarlane** [comment]
there's an effort underway right now to use the generated vso-node-api which does all the versioning etc... like standard clients.

**Comment by bryanmacfarlane** [comment]
Might want to get that to land which is close - Teddy Ward is working on it.

File: src/agent/api/timelineapi.ts [hunk 1]
```diff
@@ -7,10 +7,14 @@ import restm = require('./restclient');
 
 export class TimelineApi implements ifm.ITimelineApi {
```
Comment (bryanmacfarlane) [inline src/agent/api/timelineapi.ts]: This is all generated now so we don't want to change this - contact Teddy Ward for that to land first

**Comment by teddyward** [comment]
watch this branch for updates on getting the generated APIs up and running:
https://github.com/Microsoft/vso-agent/tree/users/teddyward/vsonodeapi

converted/deleted the Task, Agent, and Build APIs (though not 100% tested), working on getting rid of Timeline now because those endpoints will all be in Task & TaskAgent now

File: src/agent/job.ts [hunk 2]
```diff
@@ -113,7 +113,8 @@ export class JobRunner {
                     // TODO: replace build with sender id once confirm how sent
```
Comment (bryanmacfarlane) [inline src/agent/job.ts]: the job module is getting reworked completely so we shouldn't take changes.  
https://github.com/microsoft/steprunner

File: src/agent/plugins/release/prepare.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import context = require('../../context');
+
+export function pluginName() {
+    return "Download artifacts";
+}
+
+// what shows in progress view
+export function pluginTitle() {
+    return "pluginTitle: Downloading artifacts";
```
Comment (bryanmacfarlane) [inline src/agent/plugins/release/prepare.ts]: I don't think you want "pluginTitle: " showing up in the web progress view ...

**Comment by bryanmacfarlane** [comment]
?I'll test after you finish the timeline work


________________________________

**Comment by teddyward** [comment]
Timelines done, see this commit:
https://github.com/Microsoft/vso-agent/commit/a2c6c0f18253c18b5861db26a6989d9c0ef873ab

Doing FileContainer and getting some help with TestManagement next

**Comment by bryanmacfarlane** [comment]
?FileContainer is the bar where we merge the API work.  The test management stuff can come in right after but doesn't need to block.


Wen FileContainer is in, I'll test and then we'll merge


Thanks


________________________________

**Pull request merged**